src/main/host/syscall/handler/unistd.rs

```diff
@@ -109,7 +109,11 @@ impl SyscallHandler {
         if let Some(replaced_desc) = replaced_desc {
             // from 'man 2 dup2': "If newfd was open, any errors that would have been reported at
             // close(2) time are lost"
-            CallbackQueue::queue_and_run(|cb_queue| replaced_desc.close(ctx.objs.host, cb_queue));
+            crate::utility::legacy_callback_queue::with_global_cb_queue(|| {
```

Comment:
I wonder if we can make it harder to forget to do this. Would it make sense for `queue_and_run` to set up and tear down the legacy callback queue?

Reply:
I thought about that originally, but at the time I wanted to keep them separate so that it would be easy to rip out the legacy code when we didn't have C file objects anymore, and so that this global stuff would be explicit and not a hidden side effect when using a `CallbackQueue`. But it might make sense now to put it in `CallbackQueue` anyways.

I'll try it and if it works out okay then I'll make a separate PR for that.